lsp-haskell.el

```diff
@@ -495,6 +495,26 @@ Note that this must be set to true in order to get completion of pragmas."
 ;; ---------------------------------------------------------------------
 ;; Starting the server and registration with lsp-mode
 
+(cl-defmethod lsp-clients-extract-signature-on-hover (contents (_server-id (eql lsp-haskell)))
+  "Display the type signature of the function under point."
+  (let* ((groups (--filter (s-equals? "```haskell" (car it))
+                           (-partition-by #'s-blank?
+                                          (->> (lsp-get contents :value)
```

Comment:
I hope I did this right, but this should be agnostic wrt the chosen lsp-mode configuration. Still, I've only tested it with `use-plists`, so it would be nice if someone using hashtables